src/mono/sample/wasm/browser-bench/BenchTask.cs

```diff
@@ -79,7 +89,10 @@ public Result RunBatch(BenchTask task, int milliseconds)
                 start = DateTime.Now;
                 for (i = 0; i < steps; i++)
                 {
-                    RunStep();
+                    if (HasRunStepAsync)
+                        await RunStepAsync();
```

Comment:
Note that the async continuation is probably executed from `setTimer(... , 0)` so, if there is work in that scheduler, it may impact measurement. But compared to whole emscripten startup, probably not big problem.

Reply:
Yeah. I think there should not be much else happening at that time, besides things which are part of the startup.

Not sure whether we can improve that.